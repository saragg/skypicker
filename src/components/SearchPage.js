import React, { Component } from 'react';
import SearchForm from './SearchForm';
import ListOf from './List';
import FlightItem from './FlightItem';
import ContainerMessage from './ContainerMessage';
import CircularProgress from 'material-ui/CircularProgress';
import FontIcon from 'material-ui/FontIcon'

import Api from '../services/Api';

class SearchPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      flights: [],
      noResults: false,
      loading: false,
      noSearchYet: true,
    }

    this.searchFlights = this.searchFlights.bind(this)
  }

  searchFlights = (params) => {
    this.setState({
      flights: [],
      loading: true
    })
    Api.getFlights(params)
    .then((response) => {

      this.setState({
        flights: response.data,
        noResults: response.data.length === 0,
        loading: false,
        noSearchYet: false
      })
    })
    .catch(()=>
      this.setState({
        noResults: true,
        flights: [],
        loading: false
      })
    )
  }

  render() {
    const { flights, noResults, loading, noSearchYet } = this.state

    const ResultsList = ListOf(FlightItem)

    return (
      <div>
        <div className="upper-container">
          <SearchForm searchFlights={this.searchFlights}/>
        </div>
        <div className="main-container">
          {loading && <div className="loading">
            <CircularProgress size={80} thickness={5} />
          </div>}
          {noSearchYet && <ContainerMessage icon="fa fa-globe"
            message="Make one search and start traveling!"/>
          }
          <ResultsList items={flights} noResults={noResults}/>
        </div>
      </div>
    )
  }
}

export default SearchPage;