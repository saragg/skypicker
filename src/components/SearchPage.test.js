import React from 'react';
import SearchPage from './SearchPage';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

test('Renders correctly', () => {

  const searchFlights = () => {
    return;
  }

  const tree = renderer.create(
    <MuiThemeProvider>
      <SearchPage />
    </MuiThemeProvider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});