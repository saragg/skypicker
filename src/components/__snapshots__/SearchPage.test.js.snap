// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    className="upper-container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="search-form"
      >
        <div
          className="half"
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "position": "relative",
                "width": 256,
              }
            }
          >
            <div
              className={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "cursor": "auto",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "height": 72,
                  "lineHeight": "24px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 256,
                }
              }
            >
              <label
                className={undefined}
                htmlFor="flyFrom-undefined-Origin-9652"
                onTouchTap={undefined}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.3)",
                    "lineHeight": "22px",
                    "muiPrepared": true,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "top": 38,
                    "transform": "scale(1) translate(0, 0)",
                    "transformOrigin": "left top",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "zIndex": 1,
                  }
                }
              >
                Origin
              </label>
              <input
                autoComplete="off"
                disabled={false}
                id="flyFrom-undefined-Origin-9652"
                name="flyFrom"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={false}
                style={
                  Object {
                    "WebkitOpacity": 1,
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "marginTop": 14,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
                type="text"
                value=""
              />
              <div>
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 1,
                      "borderColor": "#e0e0e0",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                />
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 2,
                      "borderColor": "#00bcd4",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "transform": "scaleX(0)",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
            <div
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <div
            style={
              Object {
                "display": "inline-block",
                "muiPrepared": true,
                "position": "relative",
                "width": 256,
              }
            }
          >
            <div
              className={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "cursor": "auto",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "height": 72,
                  "lineHeight": "24px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 256,
                }
              }
            >
              <label
                className={undefined}
                htmlFor="to-undefined-Destination-31899"
                onTouchTap={undefined}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.3)",
                    "lineHeight": "22px",
                    "muiPrepared": true,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "top": 38,
                    "transform": "scale(1) translate(0, 0)",
                    "transformOrigin": "left top",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "zIndex": 1,
                  }
                }
              >
                Destination
              </label>
              <input
                autoComplete="off"
                disabled={false}
                id="to-undefined-Destination-31899"
                name="to"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={false}
                style={
                  Object {
                    "WebkitOpacity": 1,
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "marginTop": 14,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
                type="text"
                value=""
              />
              <div>
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 1,
                      "borderColor": "#e0e0e0",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                />
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 2,
                      "borderColor": "#00bcd4",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "transform": "scaleX(0)",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
            <div
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
        </div>
        <div
          className="half"
        >
          <div
            className={undefined}
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            <div
              className={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "cursor": "auto",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "height": 72,
                  "lineHeight": "24px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 256,
                }
              }
            >
              <label
                className={undefined}
                htmlFor="dateFrom-undefined-Departuredate-24913"
                onTouchTap={undefined}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.3)",
                    "lineHeight": "22px",
                    "muiPrepared": true,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "top": 38,
                    "transform": "scale(1) translate(0, 0)",
                    "transformOrigin": "left top",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "zIndex": 1,
                  }
                }
              >
                Departure date
              </label>
              <input
                disabled={false}
                id="dateFrom-undefined-Departuredate-24913"
                name="dateFrom"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitOpacity": 1,
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "marginTop": 14,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
                type="text"
                value=""
              />
              <div>
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 1,
                      "borderColor": "#e0e0e0",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                />
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 2,
                      "borderColor": "#00bcd4",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "transform": "scaleX(0)",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
            </div>
          </div>
          <div
            className={undefined}
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            <div
              className={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "cursor": "auto",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "height": 72,
                  "lineHeight": "24px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 256,
                }
              }
            >
              <label
                className={undefined}
                htmlFor="dateTo-undefined-Returndate-63986"
                onTouchTap={undefined}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.3)",
                    "lineHeight": "22px",
                    "muiPrepared": true,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "top": 38,
                    "transform": "scale(1) translate(0, 0)",
                    "transformOrigin": "left top",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "zIndex": 1,
                  }
                }
              >
                Return date
              </label>
              <input
                disabled={false}
                id="dateTo-undefined-Returndate-63986"
                name="dateTo"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitOpacity": 1,
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "marginTop": 14,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
                type="text"
                value=""
              />
              <div>
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 1,
                      "borderColor": "#e0e0e0",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                />
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": 2,
                      "borderColor": "#00bcd4",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "transform": "scaleX(0)",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                     0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "minWidth": 88,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "#ff4081",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": "100%",
                "zIndex": 1,
              }
            }
            tabIndex={0}
            type="Submit"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                style={
                  Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <span
                  className="fa fa-search"
                  color="#ffffff"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "color": "#ffffff",
                      "display": "inline-block",
                      "fontSize": 24,
                      "marginLeft": 12,
                      "marginRight": 0,
                      "muiPrepared": true,
                      "position": "relative",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "userSelect": "none",
                      "verticalAlign": "middle",
                    }
                  }
                />
                <span
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "margin": 0,
                      "muiPrepared": true,
                      "opacity": 1,
                      "paddingLeft": 8,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "userSelect": "none",
                    }
                  }
                >
                  Search
                </span>
              </div>
            </div>
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    className="main-container"
  >
    <div
      className="container-message"
    >
      <span
        className="fa fa-globe"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#999",
            "display": "inline-block",
            "fontSize": "90px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      />
      <p>
        Make one search and start traveling!
      </p>
    </div>
    <div />
  </div>
</div>
`;
